IoTuring/Entity/Deployments/Notify/Notify.py

```diff
@@ -1,49 +1,68 @@
 from IoTuring.Entity.Entity import Entity
 from IoTuring.Entity.EntityData import EntityCommand
 from IoTuring.MyApp.App import App
-
 from IoTuring.Configurator.MenuPreset import MenuPreset
+from IoTuring.Entity import consts
 
-import os
+import json
 
-from IoTuring.Entity import consts
+import os
 
 supports_win = True
 try:
-    import win10toast
+    import tinyWinToast.tinyWinToast as twt
 except:
     supports_win = False
 
+commands = {
+    consts.OS_FIXED_VALUE_LINUX: 'notify-send "{}" "{}"',
+    consts.OS_FIXED_VALUE_MACOS: 'osascript -e \'display notification "{}" with title "{}"\''
+}
 
-supports_unix = True
-try:
-    import notify2
-except:
-    supports_unix = False
 
 KEY = 'notify'
 
 # To send notification data through message payload use these two
 PAYLOAD_KEY_TITLE = "title"
 PAYLOAD_KEY_MESSAGE = "message"
+PAYLOAD_SEPARATOR = "|"
 
 CONFIG_KEY_TITLE = "title"
 CONFIG_KEY_MESSAGE = "message"
 
-DEFAULT_DURATION = 10  # Seconds
+ICON_FILENAME = "icon.png"
 
+MODE_DATA_VIA_CONFIG = "data_via_config"
+MODE_DATA_VIA_PAYLOAD = "data_via_payload"
 
 class Notify(Entity):
     NAME = "Notify"
     DEPENDENCIES = ["Os"]
     ALLOW_MULTI_INSTANCE = True
 
+    # Data is set from configurations if configurations contain both title and message
+    # Otherwise, data is set from payload (even if only one of title or message is set)
     def Initialize(self):
         try:
             self.config_title = self.GetConfigurations()[CONFIG_KEY_TITLE]
             self.config_message = self.GetConfigurations()[CONFIG_KEY_MESSAGE]
+            self.data_mode = MODE_DATA_VIA_CONFIG
         except Exception as e:
-            raise Exception("Configuration error: " + str(e))
+            self.data_mode = MODE_DATA_VIA_PAYLOAD
+
+        if self.data_mode == MODE_DATA_VIA_CONFIG:
+            if not self.config_title or not self.config_message:
+                self.data_mode = MODE_DATA_VIA_PAYLOAD
```

Comment:
The problem with this check, if there is some problem with the configuration, it will fall back to payload notification, and the user expects that something happens calling the command without a payload, but nothing happens, because we are waiting for a payload.

I think a solution could be to have a separate question on configuration flow _before_ the title and message, if they want to use payload or config. I think relying only if the config string is empty is not enough, considering the different types of nothing (e.g. empty string vs `Null`)

Reply:
Yes, what if we warn the user if something bad happens with configuration ? 
Because a question to use a method or the other one would need additional features for the MenuPreset..

Reply:
I added ab extra check at f40cc6ddc50bd6af599f0ca6ff7b8cedf3c14e56 I think that covers most possible errors.